app/services/articles/updater.rb

```diff
@@ -12,6 +12,9 @@ def self.call(*args)
     end
 
     def call
+      rate_limiter = RateLimitChecker.new(user)
+      raise if rate_limiter.limit_by_action("article_update")
```

Comment:
I think it'd be good to raise a more specific exception than `RuntimerError`. Maybe it's time to add a generic `RateLimitError` in `app/errors` which we then raise with a specific message:

```ruby
action = "article_update"
rate_limiter = RateLimitChecker.new(user)
limit_exceeded = rate_limiter.limit_by_action("article_update")
raise RateLimitError, action if limit_exceeded
```

Since the service object is quite specific, you can probably even pull the action name out into a constant.